docs/user-guide/configuring/configuration-files.md

```diff
@@ -0,0 +1,394 @@
+In the same way, if there is a `package.json` file in the root directory with an `eslintConfig` field, the configuration it describes will apply to all subdirectories beneath it, but the configuration described by the `.eslintrc` file in the tests directory will override it where there are conflicting specifications.
+
+```text
+your-project
+├── package.json
+├── lib
+│ └── source.js
+└─┬ tests
+  ├── .eslintrc
+  └── test.js
+```
+
+If there is an `.eslintrc` and a `package.json` file found in the same directory, `.eslintrc` will take a priority and `package.json` file will not be used.
+
+By default, ESLint will look for configuration files in all parent folders up to the root directory. This can be useful if you want all of your projects to follow a certain convention, but can sometimes lead to unexpected results. To limit ESLint to a specific project, place `"root": true` inside the `eslintConfig` field of the `package.json` file or in the `.eslintrc.*` file at your project's root level. ESLint will stop looking in parent folders once it finds a configuration with `"root": true`.
+
+```js
+{
+    "root": true
+}
+```
+
+And in YAML:
+
+```yaml
+---
+  root: true
+```
+
+For example, consider `projectA` which has `"root": true` set in the `.eslintrc` file in the `lib/` directory.  In this case, while linting `main.js`, the configurations within `lib/` will be used, but the `.eslintrc` file in `projectA/` will not.
+
+```text
+home
+└── user
+    └── projectA
+        ├── .eslintrc  <- Not used
+        └── lib
+            ├── .eslintrc  <- { "root": true }
+            └── main.js
+```
+
+The complete configuration hierarchy, from highest precedence to lowest precedence, is as follows:
+
+1. Inline configuration
+    1. `/*eslint-disable*/` and `/*eslint-enable*/`
+    1. `/*global*/`
+    1. `/*eslint*/`
+    1. `/*eslint-env*/`
+1. Command line options (or CLIEngine equivalents):
+    1. `--global`
+    1. `--rule`
+    1. `--env`
+    1. `-c`, `--config`
+1. Project-level configuration:
+    1. `.eslintrc.*` or `package.json` file in the same directory as linted file
+    1. Continue searching for `.eslintrc` and `package.json` files in ancestor directories (parent has the highest precedence, then grandparent, etc.), up to and including the root directory or until a config with `"root": true` is found.
+
+## Extending Configuration Files
+
+A configuration file can extend the set of enabled rules from base configurations.
+
+The `extends` property value is either:
+
+* a string that specifies a configuration (either a path to a config file, the name of a shareable config, `eslint:recommended`, or `eslint:all`)
+* an array of strings: each additional configuration extends the preceding configurations
+
+ESLint extends configurations recursively, so a base configuration can also have an `extends` property. Relative paths and shareable config names in an `extends` property are resolved from the location of the config file where they appear.
```

Comment:
It seems like we are missing an important piece of information here: the way that config names are resolved. So `example` is resolved to `eslint-config-example`, for example.